cmd/modern/root/install/mssql-base.go

```diff
@@ -468,32 +477,55 @@ CHECK_POLICY=OFF`
 	}
 }
 
+func getEscapedDbName(dbName string) string {
+	dbName = strings.ReplaceAll(dbName, "'", "''")
+	dbName = strings.ReplaceAll(dbName, "]", "]]")
+	return dbName
+}
+
+func parseDbName(usingDbUrl string) string {
+	u, _ := url.Parse(usingDbUrl)
+	dbToken := path.Base(u.Path)
+	tokens := strings.Split(dbToken, ".bak,")
+	if len(tokens) > 1 {
+		return tokens[1]
+	}
+	return ""
+}
+
 func (c *MssqlBase) downloadAndRestoreDb(
 	controller *container.Controller,
 	containerId string,
 	userName string,
 ) {
 	output := c.Cmd.Output()
+	databaseName := parseDbName(c.usingDatabaseUrl)
```

Comment:
>parseDbName(c.usingDat [](http://example.com/codeflow?start=17&length=22)

it feels like there's an opportunity to consolidate some of this code with the validation code by changing parseDbName to return the database name for both cases - when there's a `,dbname` and when there isn't.
Then the unit test for parseDbName can validate the full range of  URLs with and without escaped characters in the file names in the url etc.

Reply:
Consolidated the code to set dbName to filename when dbName is not specified.